src/constraint/spec.ts

```diff
@@ -378,6 +380,11 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
     allowEnumSpecForProperties: false,
     strict: false,
     satisfy: (specM: SpecQueryModel, schema: Schema, opt: QueryConfig) => {
+      if (specM.enumSpecIndex.hasProperty(Property.CHANNEL) &&
```

Comment:
I don't understand why you add this condition.  Looks like a hack.  
At least you should add comment justifying your design decision (if it's well principled, not a hack.)